test/libsolidity/semanticTests/viaYul/memory_struct_allow.sol

```diff
@@ -0,0 +1,21 @@
+contract C {
+    struct S {
+        uint256 a;
+        uint256 b;
+    }
+
+    function f() public pure returns (uint256 a, uint256 b){
+        assembly {
+            // Make free memory dirty to check that the struct allocation cleans it up again.
+            let freeMem := mload(0x40)
+            mstore(freeMem, 42)
+            mstore(add(freeMem, 32), 42)
```

Comment:
Here we are storing `42` at memory address `0x80 + 0x20 = 0xa0`

Reply:
Note to self: We don't update free memory pointer to hold `0xa0 + 0x20 = 0xc0` so it still points to `0x80`

Reply:
So technically, we expect `S` to not hold dirty bits stored at addresses `0x80` and `0xa0` respectively `42` in both cases.